### PR TITLE
Feat/ai reprocess

### PR DESCRIPTION
**Pull request by jamiemonksuk**
This provides extendable admin tooling to create AI generated content for existing records.

It's hooked in to the admin controller as per import and export. 
The data query uses the export query by default but can be extended for custom joins and such on a module basis.
The column headings for both export and AI tools are now built from the results, not just the base table, which was a long standing issue.

AI prompts no longer need to be present in the original data, they can be built on the fly, and any field present in the data query can be injected into the prompt using {{ twig_style_templating }}

The query for getting export previews has been split out so we can use common component for AI preview interfaces too.

The AI post import method has been refined to build its data set internally, which is much cleaner for the calling code. 
In most cases now you can just pass the multiedit from the admin form straight in.

Once processed into the AI queue, a worker job is fired as per the import tooling.

image

### CODE REVIEW

File: src/sprout/Controllers/Admin/ManagedAdminController.php [hunk 1]
```diff
@@ -400,14 +416,104 @@ public function _getExport()
...
+
+        } else {
+            $itemlist = new Itemlist();
+            $itemlist->main_columns = $cols;
+            $itemlist->items = $items;
+            $export->itemlist = $itemlist->render();
+        }
+
+        return array(
+            'title' => 'Export for ' . Enc::html(strtolower($this->friendly_name)),
```
Comment (aitken85) [inline src/sprout/Controllers/Admin/ManagedAdminController.php:434]: I'm pretty sure title is encoded within the view, so you're double encoding here!

Comment (jamiemonksuk) [reply src/sprout/Controllers/Admin/ManagedAdminController.php:434]: I think you're right. It was already like this, I just moved it, but I'm gonna fix it!

File: src/sprout/views/admin/generic_content_ai.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    'format' => 'csv',
+]);
+
+echo $refine;
+?>
+
+<h3>Preview of records</h3>
+<div class="scrollable-table"><?php echo $itemlist; ?></div>
+
+<form action="SITE/admin/ai_reprocess_action/<?php echo $controller_name; ?>" method="post">
```
Comment (aitken85) [inline src/sprout/views/admin/generic_content_ai.php:31]: Probably should encode this var.

File: src/sprout/views/admin/generic_import_ai.php [hunk 3]
```diff
@@ -76,6 +97,34 @@ function connectAiClasses($div,data,idx) {
...
+            <?php
+            Form::nextFieldDetails('Prompt text', true, 'This is the prompt we will use to generate content');
+            echo Form::multiline('m_prompt_text', ['rows' => 10]);
+            ?>
+        </div>
+        <div class="column column-6">
+            <h3>Database field options</h3>
+            <ul class="ai_db_cols">
+                <?php foreach ($db_columns as $col_name => $col_label): ?>
+                    <li><a href="javascript:;" class="js--insert-db-col" data-col="<?php echo $col_name; ?>"><?php echo $col_label; ?></a></li>
```
Comment (aitken85) [inline src/sprout/views/admin/generic_import_ai.php:121]: Same with these. `Enc::html`

**Review by aitken85 — APPROVED** [review]

**Pull request merged**